x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/filter/RestGetFiltersAction.java

```diff
@@ -55,7 +56,10 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
             getListRequest.setPageParams(
                     new PageParams(restRequest.paramAsInt(PageParams.FROM.getPreferredName(), PageParams.DEFAULT_FROM),
                     restRequest.paramAsInt(PageParams.SIZE.getPreferredName(), PageParams.DEFAULT_SIZE)));
+        } else {
+            getListRequest.setPageParams(null);
         }
+        getListRequest.setAllowNoResources(restRequest.paramAsBoolean(ALLOW_NO_RESOURCES, true));
```

Comment:
This looks like it has moved beyond refactoring to adding new functionality.  I would say that this PR should just hardcode `true` here, as that's what the behaviour was before.  (If we were to have a parameter to say finding no filters was unacceptable then maybe we'd want to call it `allow_no_filters`.)